src/AddProjectsPage/AddProjectsPage.tsx

```diff
@@ -39,6 +43,27 @@ class AddProjectsPage extends React.Component<
       preview: null,
       files: null
     };
+
+    var projectToUpdate;
+
+    if (this.props.addOrUpdateProject !== null) {
+      // triggers a call to replace projectState with oneProject by Id
+      this.props.getOneProject(this.props.addOrUpdateProject);
+      projectToUpdate = this.props.projects!;
+    }
+
+    var filledState = Object.assign({}, projectToUpdate, {
+      categoryPlaceholder: 'Choose A Category',
+      tagPlaceholder: 'Choose Some Tags',
+      teamPlaceholder: 'Add Teammates',
+      statusPlaceholder: 'Status of Project',
+      preview: null,
+      files: null
+    });
+
+    this.props.addOrUpdateProject === null
```

Comment:
Better to have the fetching done in `componentDidMount` and have have the state updated there based on results of the call.